src/packager.ts

```diff
@@ -79,14 +92,92 @@ export const handler: ScheduledHandler = Sentry.wrapHandler(
 const concatAllObjects = async (
   client: S3Client,
   bucket: string,
-  prefix: string,
+  sourcePrefix: string,
+  outputPrefix: string,
+  serviceDay: string,
+  recoveryPrefix: string,
   filename: string
 ) => {
+  const prefix = path.posix.join(sourcePrefix, serviceDay);
   const tempDir = await fs.mkdtemp(path.join(os.tmpdir(), "temp-"));
   const outputPath = path.join(tempDir, filename);
   await fs.mkdir(path.dirname(outputPath), { recursive: true });
   const outputFile = await fs.open(outputPath, "w");
 
+  if (recoveryPrefix) {
+    for await (const {
+      $metadata: metadata,
+      Contents: objects,
+    } of paginateListObjectsV2(
+      { client },
+      { Bucket: bucket, Prefix: recoveryPrefix }
+    )) {
+      if (objects === undefined)
+        throw exception("BadS3Response", JSON.stringify(metadata));
+
+      for (const { Key: key } of objects) {
+        const command = new GetObjectCommand({ Bucket: bucket, Key: key });
+        const { $metadata: metadata, Body: data } = await client.send(command);
+
+        if (data === undefined)
+          throw exception("BadS3Response", JSON.stringify(metadata));
+
+        if (
+          key &&
+          key.startsWith(`ocs-saver-${outputPrefix}-1-${serviceDay}`)
+        ) {
+          const command = new GetObjectCommand({ Bucket: bucket, Key: key });
+          const { $metadata: metadata, Body: data } = await client.send(
+            command
+          );
+
+          if (data === undefined)
+            throw exception("BadS3Response", JSON.stringify(metadata));
```

Comment:
suggestion: it looks like this pattern happens a few times:
- take a Command
- await the result of sending it to `client.send()`
- if the `data` is undefined, throw an exception
- otherwise, work with the data

You could pull that all into an `async function safeSend(command)` which either throws an exception or returns `data` for further processing.